homeassistant/components/zwave_js/config_flow.py

```diff
@@ -723,3 +808,15 @@ def __init__(self, error: str) -> None:
         """Initialize error."""
         super().__init__()
         self.error = error
+
+
+def get_serial_by_id(dev_path: str) -> str:
+    """Return a /dev/serial/by-id match for given device if available."""
+    by_id = "/dev/serial/by-id"
+    if not os.path.isdir(by_id):
+        return dev_path
+
+    for path in (entry.path for entry in os.scandir(by_id) if entry.is_symlink()):
+        if os.path.realpath(path) == dev_path:
+            return path
+    return dev_path
```

Comment:
Maybe I should move this into the usb integration in a followup

Reply:
I'm going to wait for the CI to 👍  https://github.com/home-assistant/core/pull/54935, and once thats in, open a PR to break out the duplicate code into `usb` to make this smaller

Reply:
https://github.com/home-assistant/core/pull/54968